src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -238,6 +215,38 @@ static inline void sm_config_set_clkdiv_int_frac(pio_sm_config *c, uint16_t div_
             (((uint)div_int) << PIO_SM0_CLKDIV_INT_LSB);
 }
 
+static inline void calculate_clkdiv_from_float(float div, uint16_t *div_int, uint8_t *div_frac) {
```

Comment:
this is a global namespace, you should call this something like pio_calculate_clkdiv_from_float

Reply:
Ahhh, this _did_ trigger my spidey-sense, and I wondered if it should have been named `_calculate_clkdiv_from_float` to mark it as "internal use only" as you'd typically do in Python. I've now renamed it to `pio_calculate_clkdiv_from_float` as you suggested. :+1: